### PR TITLE
changes service update logic

### PR DESCRIPTION
**Pull request by andrewpmartinez**
swap from a deep compare on service definitions to deep compare on
configurations and updatedAt comparisons for the rest of the service.
Ignores posture query values.

### CODE REVIEW

File: library/ziti.c [hunk 1]
```diff
@@ -643,23 +643,35 @@ static void ziti_re_auth(ziti_context ztx) {
...
 
+// is_service_updated returns 0 if the direct service properties
+// and configurations have not been altered. Will return non-0
+// values if they have. This ignores posture query alterations.
+static int is_service_updated(ziti_service *new, ziti_service *old) {
+    if (strcmp(old->updated_at, new->updated_at) != 0) {
+        return 1;
+    }
+
+    return model_map_compare(&old->config, &new->config, get_model_map_meta());
```
Comment (ekoby) [inline library/ziti.c:672]: pretty sure it is the wrong type_meta to pass in here. It should the type_meta of the values in the map

File: library/ziti.c [hunk 2]
```diff
@@ -643,23 +643,35 @@ static void ziti_re_auth(ziti_context ztx) {
...
 }
 
+// is_service_updated returns 0 if the direct service properties
+// and configurations have not been altered. Will return non-0
+// values if they have. This ignores posture query alterations.
+static int is_service_updated(ziti_service *new, ziti_service *old) {
+    if (strcmp(old->updated_at, new->updated_at) != 0) {
+        return 1;
+    }
+
```
Comment (ekoby) [inline library/ziti.c:671]: didn't we discuss that it would also trigger on posture passing state change?

File: library/ziti.c (hunk 1 above)
Comment (andrewpmartinez) [reply library/ziti.c]: Sure is. Fixed.

File: library/ziti.c (hunk 2 above)
Comment (andrewpmartinez) [reply library/ziti.c:690]: Sure did, fixed.

File: includes/ziti/ziti_model.h [hunk 3]
```diff
@@ -80,7 +80,9 @@ XX(permissions, string, array, permissions, __VA_ARGS__) \
 XX(encryption, bool, none, encryptionRequired, __VA_ARGS__) \
 XX(perm_flags, int, none, NULL, __VA_ARGS__) \
 XX(config, json, map, config, __VA_ARGS__) \
-XX(posture_query_set, ziti_posture_query_set, array, postureQueries, __VA_ARGS__)
+XX(posture_query_set, ziti_posture_query_set, array, postureQueries, __VA_ARGS__) \
+XX(posture_query_map, ziti_posture_query_set, map, posturePolicies, __VA_ARGS__) \
```
Comment (andrewpmartinez) [inline includes/ziti/ziti_model.h:84]: Adds map version of posture sets.

File: library/posture.c [hunk 4]
```diff
@@ -236,39 +238,36 @@ void ziti_send_posture_data(ziti_context ztx) {
...
-            setIdx++;
         }
     }
 
-    // mark all responses obsolete in case they were removed
+    // mark responses obsolete in case they were removed
     pr_info *resp;
     MODEL_MAP_FOREACH(name, resp, &ztx->posture_checks->responses) {
-        resp->obsolete = true;
+        if (!resp->pending && !resp->should_send) {
```
Comment (andrewpmartinez) [inline library/posture.c:268]: This was marking process responses as obsolete before they were processed on the worker. The worker would finish and the process response would not be sent as it was marked obsolete.

File: library/posture.c [hunk 5]
```diff
@@ -327,15 +326,14 @@ void ziti_send_posture_data(ziti_context ztx) {
         }
     }
 
-    if (procInfo != NULL) {
+    if (model_map_size(&processes) > 0) {
```
Comment (andrewpmartinez) [inline library/posture.c:329]: `procInfo` would be null on subsequent runs. Causing process responses to not be submitted. We have a map of process responses now, so we can simply check that.

File: library/ziti.c [hunk 6]
```diff
@@ -461,6 +468,16 @@ void ziti_dump(ziti_context ztx, int (*printer)(void *arg, const char *fmt, ...)
         MODEL_MAP_FOREACH(cfg, cfg_json, &zs->config) {
             printer(ctx, "\tconfig[%s]=%s\n", cfg, cfg_json);
         }
+
```
Comment (andrewpmartinez) [inline library/ziti.c:471]: Added posture queries to `ziti_dump`

File: library/ziti.c [hunk 7]
```diff
@@ -882,9 +990,25 @@ static void edge_routers_cb(ziti_edge_router_array ers, const ziti_error *err, v
...
+        ztx->identity_data = data;
+    }
+
+    update_ctrl_status(ztx, FIELD_OR_ELSE(err, err, 0), FIELD_OR_NULL(err, message));
+}
+
+static void session_post_auth_query_cb(ziti_context ztx, int status, void *ctx) {
     ziti_session *session = ztx->session;
 
+    ziti_ctrl_current_identity(&ztx->controller, update_identity_data, ztx);
```
Comment (andrewpmartinez) [inline library/ziti.c:1010]: This was moved to after MFA is completed. It was 401'ing all the time on MFA identities.

**Review by ekoby — APPROVED** [review]

**Pull request merged**